### PR TITLE
Use batched memcpy when writing ORC statistics

### PR DESCRIPTION
**Pull request by vuule**
## Description
This PR replaces a set of per-column, per-rowgroup D2D memcopies with a single call to the `batched_memcpy_async` utility. Should improve performance when writing wide tables.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/orc/writer_impl.cu [hunk 1]
```diff
@@ -1386,29 +1387,34 @@ encoded_footer_statistics finish_statistic_blobs(Footer const& footer,
...
+
+      h_srcs.push_back(per_chunk_stats.stripe_stat_merge[i].device_ptr() + col * stripes_per_col);
+      h_dsts.push_back(stats_merge.device_ptr() + (num_stripes * col) + num_entries_seen);
+      h_lens.push_back(stripes_per_col * sizeof(statistics_merge_group));
     }
     num_entries_seen += stripes_per_col;
   }
 
+  auto const& mr    = cudf::get_current_device_resource_ref();
+  auto const d_srcs = cudf::detail::make_device_uvector_async<void*>(h_srcs, stream, mr);
```
Comment (mhaseeb123) [inline cpp/src/io/orc/writer_impl.cu:1412]: CTAD should be able to pick up the types, right?

**Comment by vuule** [comment]
Measured performance impact without compression and IO - average improvement of 7%. The impact is expected to shrink a lot with compression and/or IO.

File: cpp/src/io/orc/writer_impl.cu [hunk 2]
```diff
@@ -1386,29 +1387,34 @@ encoded_footer_statistics finish_statistic_blobs(Footer const& footer,
...
+      h_dsts.push_back(stats_merge.device_ptr() + (num_stripes * col) + num_entries_seen);
+      h_lens.push_back(stripes_per_col * sizeof(statistics_merge_group));
     }
     num_entries_seen += stripes_per_col;
   }
 
+  auto const& mr    = cudf::get_current_device_resource_ref();
+  auto const d_srcs = cudf::detail::make_device_uvector_async(h_srcs, stream, mr);
+  auto const d_dsts = cudf::detail::make_device_uvector_async(h_dsts, stream, mr);
+  auto const d_lens = cudf::detail::make_device_uvector_async(h_lens, stream, mr);
```
Comment (vuule) [inline cpp/src/io/orc/writer_impl.cu:1414]: Tempted to create these inline in `batched_memcpy_async`, but it seems a bit too much.

**Review by mhaseeb123 — APPROVED** [review]
Looks good!

**Review by PointKernel — APPROVED** [review]

**Pull request merged**